### PR TITLE
Slight efficiency improvement on all page transitions

### PR DESCRIPTION
**Pull request by xster**
There's a slight redundancy having the various page transitions themselves listening to the route animation since all of the rebuilding can be delegated to its descendants. 

This should effectively remove one build per frame.

### CODE REVIEW

File: packages/flutter/lib/src/material/page.dart [hunk 1]
```diff
@@ -8,38 +8,38 @@ import 'package:flutter/widgets.dart';
...
-    end: FractionalOffset.topLeft
-  );
-
+    @required Animation<double> routeAnimation,
+    @required this.child,
+  }) : _positionAnimation = _kTween.animate(new CurvedAnimation(
+         parent: routeAnimation, // The route's linear 0.0 - 1.0 animation.
+         curve: Curves.fastOutSlowIn
+       )),
+       _opacityAnimation = new CurvedAnimation(parent: routeAnimation,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/page.dart]: nit: newline before `parent`

File: packages/flutter/lib/src/material/page.dart [hunk 2]
```diff
@@ -8,38 +8,38 @@ import 'package:flutter/widgets.dart';
...
-  static final FractionalOffsetTween _kTween = new FractionalOffsetTween(
-    begin: const FractionalOffset(0.0, 0.25),
-    end: FractionalOffset.topLeft
-  );
-
+    @required Animation<double> routeAnimation,
+    @required this.child,
+  }) : _positionAnimation = _kTween.animate(new CurvedAnimation(
+         parent: routeAnimation, // The route's linear 0.0 - 1.0 animation.
+         curve: Curves.fastOutSlowIn
```
Comment (Hixie) [inline packages/flutter/lib/src/material/page.dart]: nit: trailing comma

**Comment by Hixie** [comment]
LGTM

**Comment by xster** [comment]
Just a random thought, maybe this exists already but I didn't find it in https://flutter.io/debugging

We could add a 'insight' output verbosity filter or something in debug mode where we log the number of times States get setState called on by the user but creates the same (shallow) Widget as the last build call. Then we can print a nice table to help users catch redundancies.

**Pull request merged**